src/server/chromium/crPage.ts

```diff
@@ -683,17 +684,16 @@ class FrameSession {
   }
 
   async _initBinding(binding: PageBinding) {
+    const worldName = binding.world === 'utility' ? UTILITY_WORLD_NAME : undefined;
     await Promise.all([
-      this._client.send('Runtime.addBinding', { name: binding.name }),
-      this._client.send('Page.addScriptToEvaluateOnNewDocument', { source: binding.source })
+      this._client.send('Runtime.addBinding', { name: binding.name, executionContextName: worldName }),
+      this._client.send('Page.addScriptToEvaluateOnNewDocument', { source: binding.source, worldName })
     ]);
   }
 
   async _onBindingCalled(event: Protocol.Runtime.bindingCalledPayload) {
     const context = this._contextIdToContext.get(event.executionContextId)!;
-    const pageOrError = await this._crPage.pageOrError();
-    if (!(pageOrError instanceof Error))
-      this._page._onBindingCalled(event.payload, context);
+    await this._page._onBindingCalled(event.payload, context);
```

Comment:
Why this change? I hoped we have a test for a binding in errored pages, but guess not.

Reply:
I thought it was unnecessary. I'll put it back.